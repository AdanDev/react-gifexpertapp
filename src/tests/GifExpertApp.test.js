import React from 'react';
import {GifExpertApp} from '../GifExpertApp';
import {shallow} from 'enzyme';

describe('Pruebas en <GifExpertApp />', () => {
    
    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Overwatch', 'LoR'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })

})
