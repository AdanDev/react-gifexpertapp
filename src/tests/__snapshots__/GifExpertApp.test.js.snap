// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Overwatch"
      key="Overwatch"
    />
    <GifGrid
      category="LoR"
      key="LoR"
    />
  </ol>
</div>
`;
